### PR TITLE
Fix AutoRotatingFileDestination initial maxFileSize/maxTimeInterval

### PR DESCRIPTION
**Pull request by Gucky**
AutoRotationFileDestionation did not properly use maxFileSize from initializer and calculation/use of maxTimeInterval did not happen as expected. Also added default parameter for maxTimeInterval to initializer to be able to configure this value from start on.
also see issue https://github.com/DaveWoodCom/XCGLogger/issues/201 by vronin

### CODE REVIEW

File: Sources/XCGLogger/Destinations/AutoRotatingFileDestination.swift [hunk 1]
```diff
@@ -251,7 +256,10 @@ open class AutoRotatingFileDestination: FileDestination {
         // File Size
         guard currentLogFileSize < targetMaxFileSize else { return true }
 
-        // Time Interval
+        // Time Interva, zero = never rotate
```
Comment (laposheureux) [inline Sources/XCGLogger/Destinations/AutoRotatingFileDestination.swift]: Missing a few 'l's

Comment (Gucky) [inline Sources/XCGLogger/Destinations/AutoRotatingFileDestination.swift]: Ups, indeed. Thanks!

**Comment by laposheureux** [comment]
Hey @Gucky - were you able to get AutoRotatingFileDestination to work at all, even with the issues in your PR here?

**Comment by Gucky** [comment]
@laposheureux, I had to manually check for folder/file upfront and create it myself. So yes, I found the same issue you are describing in #208.
That's on my todo list to merge into the XCGLogger and also create a pull request for that.
I did not observer the double rotation. But honestly did not check it.

I guess some problems occure due to the fact that by pure initialization Destinations of all kinds directly start to perform cleanup work (e.g. rotation) already. This puts us in the position - like with the maxFileSize/maxTimeInterval - that we have no chance to customize properties before.
If initializers would not perform these kind of tasks and they would only get triggered by new arriving log lines - and they trigger them already - some that might be a better approach, but also mean a deeper refactoring.

**Comment by laposheureux** [comment]
Took a stab at the various things I ran into - in case you have any opinions on the approach given that part of it was on your list ^

**Comment by DaveWoodCom** [comment]
Merged into 5.0.2. Thanks for the PR.

**Pull request closed without merging**